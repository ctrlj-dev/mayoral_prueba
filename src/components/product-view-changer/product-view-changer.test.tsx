import { fireEvent, render } from '@testing-library/react';
import ProductViewChanger from './product-view-changer-component';

describe('ProductView Changer specs', () => {
    test('Test to check if components its rendering properly', () => {
        //Arrange
        //Act
        const screen = render(
            <div data-testid="component">
                <ProductViewChanger
                    viewMoreProductEvent={() => true}
                    viewLessProductEvent={() => false}
                />
            </div>)

        const element = screen.getByTestId('component').firstChild.nodeName;
        const elementButton = screen.getAllByRole('button');
 
        //Assert
        expect(element).not.toBeNull();
        expect(elementButton).not.toBeNull();
        expect(screen.asFragment()).toMatchSnapshot();

    })
})

