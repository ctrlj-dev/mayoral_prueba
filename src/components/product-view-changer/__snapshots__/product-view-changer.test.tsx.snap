// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductView Changer specs Test to check if components its rendering properly 1`] = `
<DocumentFragment>
  <div
    data-testid="component"
  >
    <div
      class="productViewChangerContainer"
    >
      <button
        id="showLessProductosbutton"
      >
        <svg
          viewBox="0 0 45 45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="10"
            width="45"
            y="17.5"
          />
        </svg>
      </button>
      <button
        id="showMoreProductosbutton"
      >
        <svg
          viewBox="0 0 45 45"
        >
          <rect
            height="10"
            width="45"
            y="17.5"
          />
          <rect
            height="10"
            transform="translate(0 45) rotate(-90)"
            width="45"
            y="17.5"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
